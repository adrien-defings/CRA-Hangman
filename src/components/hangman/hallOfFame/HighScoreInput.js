import PropTypes from 'prop-types'
import React, { Component } from 'react'

import './hallOfFame.css'

import { saveHOFEntry } from './HallOfFame'

class HighScoreInput extends Component {
  state = { winner : '' }

  // Arrown fn for this binding
  handleWinnerUpdate = (event) => {
    this.setState({
      winner: event.target.value.toUpperCase()
    })
  }

  // Arrown fn for this binding
  persistWinner = (event) => {
    const { guesses, word } = this.props
    event.preventDefault()
    const newEntry = {
      guesses,
      word,
      player : this.state.winner,
    }

    saveHOFEntry(newEntry, this.props.onStored)
  }

  render() {
    return (
      <form className="highScoreInput" onSubmit={this.persistWinner}>
        <div>
            <input
              type="text"
              className="form-control"
              autoComplete="given-name"
              value={this.state.winner}
              onChange={this.handleWinnerUpdate}
              placeholder="Your nickname"
              />
            <button className="btn btn-primary" type="submit">Save my score</button>
        </div>
      </form>
    )
  }
}

HighScoreInput.propTypes = {
  guesses: PropTypes.number.isRequired,
  word: PropTypes.string.isRequired,
  onStored: PropTypes.func.isRequired,
}

export default HighScoreInput