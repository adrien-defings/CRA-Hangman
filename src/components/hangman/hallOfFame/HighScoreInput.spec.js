import React from 'react'

import { shallow } from 'enzyme'
import { expect } from 'chai'

import HighScoreInput from './HighScoreInput'

describe('<HighScoreInput />', () => {
  let wrapperInput
  const guesses = 10
  const word = "fakeword"

  beforeEach(() => {
    wrapperInput = shallow(
      <HighScoreInput
        guesses={guesses}
        word={word}
        onStored={() => { }}
      />
    )
    expect(wrapperInput).to.matchSnapshot()
  })

  it('should fail if no credentials are provided', () => {
    wrapperInput.setState({ winner: "TESTPLAYER" })
    const fakeEvent = { preventDefault: () => { } }

    wrapperInput.find('.highScoreInput').simulate('submit', fakeEvent)
  })
})