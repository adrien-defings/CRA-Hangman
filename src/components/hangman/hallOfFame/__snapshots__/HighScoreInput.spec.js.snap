// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighScoreInput /> should fail if no credentials are provided 1`] = `
<form
  className="highScoreInput"
  onSubmit={[Function]}
>
  <div>
    <input
      autoComplete="given-name"
      className="form-control"
      onChange={[Function]}
      placeholder="Your nickname"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      type="submit"
    >
      Save my score
    </button>
  </div>
</form>
`;
